lib/rules/no-undef.js

```diff
@@ -110,6 +113,7 @@ module.exports = function(context) {
             variable = new escope.Variable(node.local.name, scope);
         variable.defs.push(node);
```

Comment:
This isn't the AST. It's escope's representation of the data found within it. I'm already modifying it with this workaround, but it seems like my original hack didn't correctly represent the escope structure.